src/components/AppNavigation.vue

```diff
@@ -70,36 +73,35 @@ export default {
         link: '/options',
         icon: 'mdi-cog'
       }
-    ],
-    currentPageIndex: 0
-  }),
-  computed: {
-    numOfNewMessages () {
-      return this.$store.getters['chat/totalNumOfNewMessages']
-    }
-  },
-  watch: {
-    '$route.path' () {
-      this.currentPageIndex = this.getCurrentPageIndex()
-    }
-  },
-  mounted () {
-    this.currentPageIndex = this.getCurrentPageIndex()
-  },
-  methods: {
-    getCurrentPageIndex () {
-      const currentPage = this.pages.find(page => {
+    ]
+    const currentPageIndex = ref(0)
+    const store = useStore()
+    const route = useRoute()
+    watch(route, () => {
+      currentPageIndex.value = getCurrentPageIndex()
+    })
+    const getCurrentPageIndex = () => {
+      const currentPage = pages.find(page => {
         const pattern = new RegExp(`^${page.link}`)
 
-        return this.$route.path.match(pattern)
+        return route.path.match(pattern)
       })
 
-      return this.pages.indexOf(currentPage)
+      return pages.indexOf(currentPage)
+    }
+    onMounted(() => {
+      currentPageIndex.value = getCurrentPageIndex()
+    })
+
+    return {
+      pages,
+      currentPageIndex,
+      getCurrentPageIndex,
+      numOfNewMessages: computed(() => store.getters['chat/totalNumOfNewMessages'])
```

Comment:
assign it to a `const` first and the return

`const numOfNewMessages = computed(() => store.getters['chat/totalNumOfNewMessages'])`

```js
return {
  ...,
  numOfNewMessages
}
```